src/main/java/org/opensearch/knn/index/query/KNNQuery.java

```diff
@@ -38,13 +39,12 @@ public class KNNQuery extends Query {
 
     private final String field;
     private final float[] queryVector;
-    @Getter
     private final byte[] byteQueryVector;
     private int k;
     private Map<String, ?> methodParameters;
     private final String indexName;
-    @Getter
     private final VectorDataType vectorDataType;
+    private final RescoreContext rescoreContext;
```

Comment:
In our code, the RescoreContext is wrapped in an Optional in some places but not others. It's generally a good idea to be consistent in how you use Optional. If you're using it to represent a potentially absent value

public KNNQuery(...., Optional<RescoreContext> rescoreContext) {
    this.rescoreContext = rescoreContext;
}